crates/ruma-macros/src/doc/ruma_enum_attributes.md

```diff
@@ -0,0 +1,41 @@
+<!-- Keep this comment so the content is always included as a new paragraph -->
+## Container Attributes
+
+* `#[ruma_enum(rename_all = "rule")]` - Override the string representation of all unit variants
+  by using the given `rule`. By default, the string representation uses the unit variant name,
+  with the same case. This attribute allows to vary the case of the string representation with the
+  following rules. The name of the rule always matches the transformation of the string.
+
+  * `lowercase` - Convert to lowercase. `MyVariant` becomes `myvariant`.
+  * `UPPERCASE` - Convert to uppercase. `MyVariant` becomes `MYVARIANT`.
+  * `camelCase` - Convert the first character to lowercase. `MyVariant` becomes `myVariant`.
+  * `snake_case` - Add a `_` before all uppercase characters except at the start and convert all
+    characters to lowercase. `MyVariant` becomes `my_variant`.
+  * `SCREAMING_SNAKE_CASE` - Add a `_` before all uppercase characters except at the start and
+    convert all characters to uppercase. `MyVariant` becomes `MY_VARIANT`.
+  * `kebab-case` - Add a `-` before all uppercase characters except at the start and convert all
+    characters to lowercase. `MyVariant` becomes `my-variant`.
+  * `SCREAMING-KEBAB-CASE` - Add a `-` before all uppercase characters except at the start and
+    convert all characters to uppercase. `MyVariant` becomes `MY-VARIANT`.
+  * `M_MATRIX_ERROR_CASE` - The case usually used for error codes in the Matrix specification.
```

Comment:
Instead of having special matrix types with a prefix, which is usually only used on 1 or 2 enums, we could have a way to take a prefix on the container and combine it with the other rules (which is what we already do internally).

It would be more powerful and would avoid to always add new rules.

If we do that we would need to keep `SCREAMING_SNAKE_CASE` for this variant.

Reply:
Okay, I am merging this, and will try to do that in another PR.